hack/make-rules/test-cmd-util.sh

```diff
@@ -2574,6 +2575,22 @@ run_rs_tests() {
   fi
 }
 
+run_daemonset_tests() {
```

Comment:
This is to catch the original issue https://github.com/kubernetes/kubernetes/issues/43218. Test it here because `[]` specified in yaml config will be converted to `null`. 

I have a unit test in validation. Is that not enough?